src/components/ProjectDomain/addProjectDomain.tsx

```diff
@@ -0,0 +1,225 @@
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  Select,
+  SelectTrigger,
+  SelectItem,
+  SelectValue,
+  SelectContent,
+} from "@/components/ui/select";
+import { statusType } from "@/utils/common/enum";
+import { apiHelperService } from "@/services/projectdomain";
+interface DomainData {
+  _id: string;
+  label: string;
+  description: string;
+  createdAt?: string; // Optional fields
+  createdBy?: string;
+  status?: string;
+}
+
+interface Domain {
+  _id: string;
+  label: string;
+  description: string;
+}
+
+interface AddDomainProps {
+  onAddProjectDomain: (newDomain: DomainData) => void; // Prop to pass the new domain
+}
+
+// Zod schema for form validation
+const domainSchema = z.object({
+  label: z.string().nonempty("Please enter a domain name"),
+  description: z.string().nonempty("Please enter a description"),
+  status: z.enum([statusType.active]).default(statusType.active),
+});
+
+const AddProjectDomain: React.FC<AddDomainProps> = ({ onAddProjectDomain }) => {
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [domains, setDomains] = useState<Domain[]>([]); // Use Domain type here
+  const currentUser = useSelector((state: RootState) => state.user);
+  const currentUserId = currentUser.uid;
+  const { toast } = useToast();
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<DomainData>({
+    resolver: zodResolver(domainSchema),
+    defaultValues: {
+      label: "",
+      description: "",
+      status: statusType.active,
+    },
+  });
+
+  // Fetch the list of domains from the backend
+  useEffect(() => {
+    async function fetchDomains() {
+      try {
+        const response = await apiHelperService.getAllProjectdomain();
+        if (!response.data.data) {
+          toast({
+            title: "Error",
+            description: "Failed to fetch domain data . Please try again.",
+            variant: "destructive", // Red error message
+          });
+        } else {
+          setDomains(response.data.data);
+        }
+      } catch (error) {
+        toast({
+          title: "Error",
+          description: "Failed to fetch domain data . Please try again.",
+          variant: "destructive", // Red error message
+        });
+      }
+    }
+
+    fetchDomains();
+  }, []);
+
+  // Handle form submission to add a new domain
+  const onSubmit = async (data: DomainData) => {
+    // Check if domain already exists
+    const isDomainExist = domains.some(
+      (domain) => domain.label.toLowerCase() === data.label.toLowerCase(),
+    );
+
+    if (isDomainExist) {
+      setErrorMessage(`The domain "${data.label}" already exists.`);
+      return;
+    }
+
+    try {
+      const domainDataWithUser = { ...data, createdBy: currentUserId };
```

Comment:
Why Are we creating this domainDataWithUser?

Reply:
The data needs to have the user id with it thats why we are creating this variable